myfirstcatapi/models/cat_model.py

```diff
@@ -190,3 +191,23 @@ async def delete_one(cat_id: dto.CatID) -> dto.ResultCount:
         count = result.deleted_count
 
     return dto.ResultCount(count=count)
+
+
+async def update_cat_metadata(cat_id: dto.CatID, url: dto.CatURL) -> dto.ResultCount:
+    filter = dto.CatFilter(
+        cat_id=cat_id,
+    )
+
+    try:
+        match = cat_filter_to_db_match(filter)
+    except EmptyResultsFilter:
+        return dto.ResultCount(count=0)
+
+    collection = await get_collection(_COLLECTION_NAME)
+    query = {"$set": {"url": url}}
+    result = await collection.update_one(match, query)
+
+    if result:
+        count = result.modified_count
```

Comment:
[suggestion] I do not think there should be a situation where no result is returned, so this `if-statement` is not needed (and it would actually "hide" that something bad happened by throwing yet another exception because in case `result` is falsey, `count` is not initialized and thus it will crash on something else)
```suggestion
    count = result.modified_count
```
WDYT?

Reply:
done